### PR TITLE
Updates to CI and fix tests

### PR DESCRIPTION
**Pull request by naveen521kk**
- tests: fix test warning, use absolute path
- Don't use `windows-2016` (it's nearing EOL) see #66 
- Upgrade dependencies for the build
- Don't build subprojects manually on macOS build
- Add separate test workflow and not use docker in tests
- Only upload wheels if tests pass
- Include license files in all the wheels
- Link wheels statically on macOS

### CODE REVIEW

**Review by PhilippImhof — APPROVED** [review]
LGTM. Do we have to do something about the failing build?

```
cibuildwheel 2.x no longer supports Python < 3.6. Please use the 1.x series or update CIBW_SKIP
Error: Command source packing/build_pango_mac.sh && pip install cython && cd manimpango && cythonize cmanimpango.pyx -3 -k -f && cd ../ && pip install . && pkg-config --libs pango failed with code 1. None

Error: Process completed with exit code 1.
```

**Comment by naveen521kk** [comment]
> Do we have to do something about the failing build?

Ah, sorry I didn't mark it as a draft. It's still a WIP.

**Comment by PhilippImhof** [comment]
> Ah, sorry I didn't mark it as a draft. It's still a WIP.

OK, fine. You actually seemed to even have requested a review :)

**Comment by naveen521kk** [comment]
> You actually seemed to even have requested a review :)

I guess that's because you are a Code owner here,

**Comment by naveen521kk** [comment]
@PhilippImhof This PR is ready and waiting for your review :).

File: tests/test_fonts.py
Comment (PhilippImhof) [inline tests/test_fonts.py:92]:
```suggestion
    assert "Pango-WARNING **" not in captured, "Looks like Pango raised a warning?"
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:60]:
```suggestion
      - name: Build wheels (Windows)
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:82]:
```suggestion
    name: Test wheels on ${{ matrix.os }}-${{ matrix.platform_id }}-${{ matrix.python-version }}
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:118]:
```suggestion
      - name: Install test dependencies
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:122]:
```suggestion
      - name: Install wheels
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:146]:
```suggestion
      - name: Publish release
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:224]:
```suggestion
    name: Building and testing of wheels: success
```

File: .github/workflows/build.yml
Comment (PhilippImhof) [inline .github/workflows/build.yml:227]:
```suggestion
        run: echo "Building and testing of wheels: success"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:31]:
```suggestion
echo "Fetching and applying patch for Cairo build"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:64]:
```suggestion
echo "::group::Building and installing proxy-libintl"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:70]:
```suggestion
echo "::group::Building and installing Fontconfig"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:86]:
```suggestion
echo "::group::Building and installing Glib"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:100]:
```suggestion
echo "::group::Building and installing Cairo"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:114]:
```suggestion
echo "::group::Building and installing Harfbuzz"
```

File: packing/build_pango_mac.sh
Comment (PhilippImhof) [inline packing/build_pango_mac.sh:132]:
```suggestion
echo "::group::Building and installing Pango"
```

**Review by PhilippImhof — CHANGES_REQUESTED** [review]
Very well done. I have just a few nitpicks w.r.t consistent language use.

**Comment by naveen521kk** [comment]
@PhilippImhof Thanks for the review. It would be nice if you could test the macOS wheels also https://github.com/ManimCommunity/ManimPango/suites/4557070397/artifacts/122415312

**Comment by PhilippImhof** [comment]
I currently have some environment problems on my machine and thus cannot properly test the macOS wheels. But I think we can go for it, no?

**Comment by naveen521kk** [comment]
> But I think we can go for it, no?

Yeah, I guess at least it works on Github's machine.

**Review by PhilippImhof — APPROVED** [review]
LGTM

**Comment by naveen521kk** [comment]
I'll do a release later today. I guess it should be a major bump right because of https://github.com/ManimCommunity/ManimPango/pull/65

**Pull request merged**
